frontend/packages/console-dynamic-plugin-sdk/src/extensions/topology.ts

```diff
@@ -39,7 +39,7 @@ export type TopologyDataModelFactory = ExtensionDeclaration<
     /** Priority for the factory */
     priority: number;
     /** Resources to be fetched from useK8sWatchResources hook. */
-    resources?: CodeRef<(namespace: string) => WatchK8sResources<any>>;
+    resources?: WatchK8sResources<any>;
```

Comment:
since you cant set a function without a CodeRef, i'm building the namespaced routes here
https://github.com/openshift/console/pull/9055/files#diff-281f5462e39247e7a7894ee5ad688563cf18b46d97e7707da0b660b79c236f41R17

Reply:
So why not make it a coderef?

Reply:
I talked to @glekner  about this too. I don't see an issue with this change unless the extension wants to hard code a namespace. In which case we can make sure to not set the namespace if one is already provided.

I don't think we have such a use case at this time that will be negatively affected by this.

Note that we still need to review the topology extensions for their exposed API and at that time may change things.